server/fleet/capabilities.go

```diff
@@ -0,0 +1,48 @@
+package fleet
+
+import "strings"
+
+// Capability represents a concrete feature of Fleet.
+type Capability string
+
+// CapabilityMap is an utility type to represent a set of capabilities.
+type CapabilityMap map[Capability]struct{}
+
+// PopulateFromString populates the CapabilityMap from a comma separated string.
+// Example: "foo,bar,baz" => {"foo": struct{}, "bar": struct{}, "baz": struct{}}
+func (c *CapabilityMap) PopulateFromString(s string) {
+	*c = make(CapabilityMap)
+
+	if s == "" {
+		return
+	}
+
+	for _, capability := range strings.Split(s, ",") {
+		(*c)[Capability(capability)] = struct{}{}
+	}
+}
+
+// String returns a comma separated string with the capabilities in the map.
+// Example: {"foo": struct{}, "bar": struct{}, "baz": struct{}} => "foo,bar,baz"
+func (c *CapabilityMap) String() string {
+	idx := 0
+	capabilities := make([]string, len(*c))
+	for capability := range *c {
```

Comment:
FYI: iterating a Go map is defined to be in random order. Double checking we are not depending on `String()` to be consistent with each call. Am ok with documenting this.

Reply:
I think we're ok, we are splitting the string and building a map on each end (clients and server) so order shouldn't matter. The only place where I had to account for this was during testing.